### DISCUSSION TITLE
GATT Service not working as expected on ESP-IDF v4.3.2

### DISCUSSION THREAD

**Discussion by amitsi76** [Mair/esp32-course#66] [Course Qeustions] [answered]
Hi David,

  I was going through BLE training material and was trying out the sample codes on ESP-IDF version 4.3.2.

Everything is working as expected till GAP code example.
Surprisingly, GATT Read Service example is not working as expected. Device Info characteristics is not appearing at all.

Are you aware of any such GATT service characteristics issue with this ESP-IDF version?

One generic question on VS studio code -> Is it possible to configure ESP-IDF extension to work with multiple versions of ESP-IDF installed locally?

Regards,
Amit

**Comment by Mair** [comment, marked as answer]
Hi Amit,
I'll need to find some time to try the code on v4.3.2
In so far as multiple IDF extensions is concerned. It is possible to have multiple versions of the IDF installed in different directories but each time you want to switch out you will need to use the IDF extension settings page and re-point the extension to the idf directory you want. 
You might also get an error when using the command line that you need to run install.bat located in the idf directory. Once you do that, you should be on that version.
Another option is to use docker

**Reply by amitsi76** [reply]
Hi Mair,

  Thanks for your response. It was helpful.

Strangely, it seems there was some issue with the dev kit board I was using. I tried the same code on the another spare unit and it worked fine.

Regards,
Amit
